source/agora/test/Base.d

```diff
@@ -99,6 +100,25 @@ shared static this()
     Runtime.extendedModuleUnitTester = &customModuleUnitTester;
 }
 
+void testAssertHandler (string file, ulong line, string msg) nothrow
+{
+    printf("[%.*s:%llu]================Assertion thrown during test: %.*s\n",
+            cast(int) file.length, file.ptr, line, cast(int) msg.length, msg.ptr);
+    try
+    {
+        scope output = stdout.lockingTextWriter();
+        CircularAppender!()().print(output);
+    }
+    catch (Exception exc)
+    {
+        scope exc_name = typeid(exc).name;
```

Comment:
I'm not sure `scope` has any effect here (it only has an effect when you `new` things). But otherwise LGTM.

Reply:
It does not, but just wanted to note that there's no escape happening.

Reply:
> It does not

It would be nice if it did though.